stdlib/public/SwiftShims/MetadataSections.h

```diff
@@ -0,0 +1,63 @@
+//===--- MetadataSections.h -----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the declaration of the MetadataSectionsRange and 
+/// MetadataSections struct, which represent, respectively,  information about  
+/// an image's section, and an image's metadata information (which is composed
+/// of multiple section information).
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_STDLIB_SHIMS_METADATASECTIONS_H
+#define SWIFT_STDLIB_SHIMS_METADATASECTIONS_H
+
+#include "SwiftStddef.h"
+#include "SwiftStdint.h"
+
+#ifdef __cplusplus
+namespace swift { 
+extern "C" {
+#endif
+
+typedef struct MetadataSectionRange {
+  __swift_uintptr_t start;
+  __swift_size_t length;
+} MetadataSectionRange;
+
+struct MetadataSections {
```

Comment:
Can you please add doxygen comments to each type declaration (and any non-obvious members)?

Reply:
I added a doxygen comment for each declaration, although I'm not sure if they're very useful right now. 

@compnerd since you wrote these data structures, do you think I should add something more to the documentation?

Reply:
Minor nit: the doxygen comments in this repository uses `///` for the comment leader.
I would describe them as:

```
/// Specifies the address range corresponding to a section.
```

and

```
/// Identifies the address space ranges for the Swift metadata required by the Swift runtime.
```

The two fields that are are "non-obvious" are `next` and `prev`:

```
/// `next` and `prev` are used by the runtime to construct a
/// circularly doubly linked list to quickly iterate over the metadata
/// from each image loaded into the address space.  These are invasive
/// to enable the runtime registration, which occurs at image load time, to
/// be allocation-free as it is invoked from an image constructor function
/// context where the system may not yet be ready to perform allocations.
/// Additionally, avoiding the allocation enables a fast load operation, which
/// directly impacts application load time.
```